managed/services/telemetry/config.go

```diff
@@ -183,9 +187,41 @@ func (c *ServiceConfig) Init(l *logrus.Entry) error { //nolint:gocognit
 		}
 	}
 
+	testClickhouseDSN := c.testClickhouseDSN()
+	if testClickhouseDSN != "" {
+		c.l.Debugf("Overriding Telemetry.DataSources.QanDBSelect.DSN with environment variables [%s][%s][%s][%s] to %q.", envClickhouseAddr, envClickHouseDatabase, envClickhouseBlockSize, envClickhousePoolSize, testClickhouseDSN)
+		c.DataSources.QanDBSelect.DSN = testClickhouseDSN
+	}
+
 	return nil
 }
 
+func (c *ServiceConfig) testClickhouseDSN() string {
+	var (
+		addr     = os.Getenv(envClickhouseAddr)
+		database = os.Getenv(envClickHouseDatabase)
+	)
+	if addr == "" {
+		return ""
+	}
+
+	if database == "" {
+		database = "pmm"
+	}
+
+	_, blockSizeOk := os.LookupEnv(envClickhouseBlockSize)
+	if blockSizeOk {
+		c.l.Warnf("Environment variable [%s] not available in github.com/clickhouse/clickhouse-go/v2", envClickhouseBlockSize)
+	}
+
+	_, poolSizeOk := os.LookupEnv(envClickhousePoolSize)
+	if poolSizeOk {
+		c.l.Warnf("Environment variable [%s] not available in github.com/clickhouse/clickhouse-go/v2", envClickhousePoolSize)
+	}
+
+	return "tcp://" + addr + "/" + database
```

Comment:
can we avoid manual URL construction and use URL or something else?

Reply:
in documentation we have comment "not for production" and similar string concat logic exists in [github.com/percona/qan-api2 main.go](https://github.com/percona/qan-api2/blob/500663da2b33bf36b94febcba56f86b07effdcdc/main.go#L321):
```go

const (
	defaultDsnF     = "clickhouse://%s?database=%s&block_size=%s&pool_size=%s"
)
```
```go
	dsnF := kingpin.Flag("dsn", "ClickHouse database DSN. Can be override with database/host/port options").Default(defaultDsnF).String()
	clickHouseDatabaseF := kingpin.Flag("clickhouse-name", "Clickhouse database name").Default("pmm").Envar("PERCONA_TEST_PMM_CLICKHOUSE_DATABASE").String()
	clickhouseAddrF := kingpin.Flag("clickhouse-addr", "Clickhouse database address").Default("127.0.0.1:9000").Envar("PERCONA_TEST_PMM_CLICKHOUSE_ADDR").String()
	clickhouseBlockSizeF := kingpin.Flag("clickhouse-block-size", "Number of rows that can be load from table in one cycle").
		Default("10000").Envar("PERCONA_TEST_PMM_CLICKHOUSE_BLOCK_SIZE").String()
	clickhousePoolSizeF := kingpin.Flag("clickhouse-pool-size", "Controls how much queries can be run simultaneously").
		Default("2").Envar("PERCONA_TEST_PMM_CLICKHOUSE_POOL_SIZE").String()
```
```go
	if *dsnF == defaultDsnF {
		dsn = fmt.Sprintf(defaultDsnF, *clickhouseAddrF, *clickHouseDatabaseF, *clickhouseBlockSizeF, *clickhousePoolSizeF)
	} else {
		dsn = *dsnF
	}
```